myFitnessTrainer/src/screens/WorkoutScreen.js

```diff
@@ -1,108 +1,198 @@
 
-const WorkoutScreen = () => {
-    const workoutLength = testExercisesDay0.length;
+    // currentExerciseStats represents the stats of the current exercise on the card
+    // currentWorkoutStats represents the the combined stats of every card in the daily workout
+    const workoutLength = dailyExerciseSet.length;
+    const [currentExerciseStats, setCurrentExerciseStats] = useState("");
     const [exerciseIndex, setExerciseIndex] = useState(0);
-    const [progress, setProgress] = useState(0/workoutLength);
-    const height = Dimensions.get('window').height;
+    const [progress, setProgress] = useState(0 / workoutLength);
+    const height = Dimensions.get("window").height;
+
+    const cardIndexRef = useRef(null);
+
+    // when exerciseIndex state changes, then we will fetch the new default exercise stats from DB
+    // currentExerciseStats state will reflect the stats of the current exrercise shown on the card
+    useEffect(() => {
+        // the last card is not a valid exercise
+        if (exerciseIndex < workoutLength - 1) {
+            getAllDocuments("exercise_stats").then((retrievedDocuments) => {
+                const exerciseStatRef =
+                    dailyExerciseSet[exerciseIndex]["default_exercise_stat_id"];
+                const targetStatId = exerciseStatRef["path"].split("/")[1];
+
+                const filteredStats = Object.keys(retrievedDocuments).filter(
+                    (statId) => statId === targetStatId
+                );
+                setCurrentExerciseStats(retrievedDocuments[filteredStats]);
+            });
+        }
+    }, [exerciseIndex]);
 
     const progressFooter = (
         <View style={styles.bottomProgressContainer}>
             <View style={styles.progressBarContainer}>
                 <Progress.Bar
                     progress={progress}
                     pas
-                    width={width * .8}
+                    width={width * 0.8}
                     height={20}
-                    color='#4682B4'
+                    color="#4682B4"
                 />
             </View>
-            <Text style={styles.progressText}>{exerciseIndex} of {workoutLength} exercises completed</Text>
+            <Text style={styles.progressText}>
+                {exerciseIndex} of {workoutLength} exercises completed
+            </Text>
         </View>
     );
 
-    const renderCarouselItem = ({index}) => {
+    const submitStatsToDB = async (index) => {
+        const exercise = dailyExerciseSet[exerciseIndex];
+        const exerciseId = await getDocumentId(
+            "exercises",
+            "name",
+            exercise["name"]
+        );
+
+        const userId = await getDocumentId(
+            "users",
+            "email",
+            auth.currentUser.email
+        );
+
+        const postObject = {
+            completed: true,
+            date: Date.now(),
+            duration: null,
```

Comment:
We can remove the 'duration' field because we will keep track of the time it took for the user to complete the exercise in their 'exercise_stat' attached to this object.

Reply:
This was a part of our original DB design but I think it is not needed with how we set up the exercise_stat